File: python/robot-name/robot_name.py
import random
import string
from time import time


class Robot(object):
    name_list = []

    def __init__(self):
        self.reset()

    def reset(self):
        random.seed(time())
        name = self._generate_name()
        if name not in self.name_list:
            self.name_list.append(name)
            self.name = name
        else:
            self.reset()

    def _generate_name(self):
        name = ""
        for _ in range(2):
            name += random.choice(string.ascii_uppercase)
        for _ in range(3):
            name += random.choice(string.digits)

        return name

File: fsharp/robot-name/RobotName.fs
module RobotName

let rnd = System.Random()
let letters = [| 'A'..'Z' |]
let lettersLen = Array.length letters - 1

let nameGenerator() =
    let number = rnd.Next(999)
    let letter () = letters.[rnd.Next(lettersLen)]

    sprintf "%c%c%03i" (letter ()) (letter ()) number

let mkRobot() = nameGenerator()
let name robot = robot
let reset robot = nameGenerator()
